drivers/mmcsd/mmcsd_sdio.c

```diff
@@ -956,13 +958,16 @@ static void mmcsd_decode_scr(FAR struct mmcsd_state_s *priv, uint32_t scr[2])
    *   DATA_STATE_AFTER_ERASE 55:55 1-bit erase status
    *   SD_SECURITY            54:52 3-bit SD security support level
    *   SD_BUS_WIDTHS          51:48 4-bit bus width indicator
-   *   Reserved               47:32 16-bit SD reserved space
+   *   Reserved               47:34 14-bit SD reserved space
+   *   CMD_SUPPORT            33:32 2-bit command supported (bit33 for cmd23)
    */
 
 #ifdef CONFIG_ENDIAN_BIG  /* Card transfers SCR in big-endian order */
   priv->buswidth     = (scr[0] >> 16) & 15;
+  priv->cmd23support =  scr[0]        & 2;
 #else
-  priv->buswidth     = (scr[0] >> 8) & 15;
+  priv->buswidth     = (scr[0] >> 8)  & 15;
+  priv->cmd23support = (scr[0] >> 24) & 2;
```

Comment:
If it is true that MMC cards ALWAYS support CMD23, do the following to simplify your patch. 

```
#ifdef CONFIG_MMCSD_MMCSUPPORT
  priv->cmd23support = 1;
#endif
```

Reply:
@ldube   cmd23support is a flag only for sd card not for mmc, in other words, mmc cards do not have scr registers

Reply:
can't do this:
> ```
> #ifdef CONFIG_MMCSD_MMCSUPPORT
>   priv->cmd23support = 1;
> #endif
> ```
since the card may be sd type not mmc type.

Reply:
> can't do this:
> 
> > ```
> > #ifdef CONFIG_MMCSD_MMCSUPPORT
> >   priv->cmd23support = 1;
> > #endif
> > ```
> 
> since the card may be sd type not mmc type.

Check if it is mmc and then do it :)

Reply:
MMC card doesn't check cmd23support, so we don't check the card type before initializing it.